source/common/common/utility.h

```diff
@@ -18,6 +18,15 @@
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
+
+/**
+ * Retrieve string description of error code, wraps strerror on POSIX systems and FormatMessage on
+ * Windows
+ * @param int error code
+ * @return const std::string error detail description
+ */
+const std::string errorDetails(int error_code);
```

Comment:
I'm not super experienced with clang-tidy and couldn't find a check that could do this, do you know of one? We could add to the format checks potentially instead?

Reply:
we're thinking of doing a format check instead, since we also have a couple places where we intentionally left `strerror` b/c there is another refactor required before we get rid of it (the filesystem impl on Windows uses the POSIX subsystem which is problematic for error handling purely Windows errors), we can use clang-format off/on to guard around those blocks that need it still

Reply:
https://github.com/envoyproxy/envoy/issues/11655 for the filesystem impl issue

Reply:
/wait

adding checks for strerror in clang format

Reply:
added a new format check, please take a look!